src/Grok.Net/Grok.cs

```diff
@@ -156,20 +189,26 @@ private void ProcessPatternLine(string line)
             {
                 return;
             }
-            try
-            {
-                Regex.Match("", strArray[1]);
-            }
-            catch
+
+            if (!ValidatePattern(strArray[1]))
             {
                 return;
             }
+            AddPatternIfNotExists(strArray[0], strArray[1]);
+        }
 
-            // check before adding to avoid an exception in case the same pattern is present in the custom patterns file.
-            if (!_patterns.ContainsKey(strArray[0]))
+        private bool ValidatePattern(string pattern)
+        {
+            try
+            {
+                _ = Regex.Match("", pattern);
+            }
+            catch(Exception e)
             {
-                _patterns.Add(strArray[0], strArray[1]);
+                _patternViolations.Add((pattern, e.Message));
```

Comment:
>  I don't see real use cases. Could you please describe any?

Well, throwing Exceptions in a constructor just did not feel good to me. But since it is already part of `Grok` in other parts of the code, I changed my mind about this. 

I changed this to throw an explicit `FormatException` having an `innerException` with the original message. I hope this is ok?